src/components/graph/layout.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import dagre from 'dagre';
+import { isNode } from 'react-flow-renderer';
+
+const nodeWidth = 150;
+const nodeHeight = 50;
+
+export function getLayoutedElements(elements) {
+    const dagreGraph = new dagre.graphlib.Graph();
+    dagreGraph.setDefaultEdgeLabel(() => ({}));
+    dagreGraph.setGraph({ direction: 'TB', align: 'UL' });
+
+    elements.forEach((el) => {
+        if (isNode(el)) {
+            dagreGraph.setNode(el.id, { width: nodeWidth, height: nodeHeight });
+        } else {
+            dagreGraph.setEdge(el.source, el.target);
+        }
+    });
+
+    dagre.layout(dagreGraph, { debugTiming: true });
```

Comment:
This parameter allows to get logs about layouting methods duration. Might be removed in a further version but still useful since network modification tree development is still in progress.